app/purchasing.cpp

```diff
@@ -398,6 +398,22 @@ void Purchasing::onFetchPurchasingPlansFinished()
 
 void Purchasing::clean()
 {
+  if ( mBackend )
+  {
+    disconnect( mBackend.get(), &PurchasingBackend::planRegistrationSucceeded, this, &Purchasing::onPlanRegistrationSucceeded );
+    disconnect( mBackend.get(), &PurchasingBackend::planRegistrationFailed, this, &Purchasing::onPlanRegistrationFailed );
+
+    disconnect( mBackend.get(), &PurchasingBackend::transactionCreationSucceeded, this, &Purchasing::onTransactionCreationSucceeded );
+    disconnect( mBackend.get(), &PurchasingBackend::transactionCreationFailed, this, &Purchasing::onTransactionCreationFailed );
+  }
+
+  if ( mMerginApi )
+  {
+    disconnect( mMerginApi, &MerginApi::apiRootChanged, this, &Purchasing::onMerginServerChanged );
+    disconnect( mMerginApi, &MerginApi::apiSupportsSubscriptionsChanged, this, &Purchasing::onMerginServerChanged );
+    disconnect( mMerginApi, &MerginApi::apiVersionStatusChanged, this, &Purchasing::onMerginServerStatusChanged );
+  }
```

Comment:
rather than connecting to MerginApi in createBackend() and disconnecting later, it is better to connect in the constructor once and then never disconnect.